src/app/pages/dashboard/widgets/apps/common/app-controls/app-controls.component.html

```diff
@@ -1,15 +1,37 @@
 <ng-container *ixWithLoadingState="app() as app">
-  @if ((app.portals | keyvalue).length > 0) {
-    <button
-      mat-icon-button
-      ixTest="apps-web-portal"
-      matTooltipPosition="above"
-      [attr.aria-label]="'Web Portal' | translate"
-      [matTooltip]="'Web Portal' | translate"
-      (click)="openWebPortal(app)"
-    >
-      <ix-icon name="mdi-web"></ix-icon>
-    </button>
+  @if (mainPortal(); as mainPortal) {
+    <div class="portal-button-wrapper">
+      <button
+        mat-icon-button
+        ixTest="apps-web-portal"
+        matTooltipPosition="above"
+        [attr.aria-label]="mainPortal.label"
+        [matTooltip]="mainPortal.label"
+        (click)="openPortal(mainPortal.url)"
+      >
+        <ix-icon name="mdi-web"></ix-icon>
+      </button>
+
+      @if (otherPortals().length > 0) {
```

Comment:
If the condition is possible where `mainPortal` doesn't exist but `otherPortals` do, because this is a nested `if`, you won't see `otherPortals` too. If the condition is not possible, you can ignore this comment.

Reply:
@RehanY147  main portal will be always set, it's `Web UI` or the first portal available if no `Web UI` available.